### PR TITLE
Revise conformance section, add Reading and Writing section

### LINKED ISSUE #89: Revise conformance section

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
Closes #89, #93.

This PR revises the conformance section and adds a Reading and Writing profiles section with clear requirements. 

Some of the  information in this new section overlap with Discovering... section and Extended profiles sections. These will require further revision (see: https://github.com/solid/webid-profile/issues/92) 

[Preview](http://htmlpreview.github.io/?https://github.com/solid/webid-profile/blob/64254c4f7854de93779cce59c6ff7bca05836fe2/index.html) | [Diff](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fsolid.github.io%2Fwebid-profile%2F&doc2=https%3A%2F%2Fraw.githubusercontent.com%2Fsolid%2Fwebid-profile%2F64254c4f7854de93779cce59c6ff7bca05836fe2%2Findex.html)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                                <code>application/ld+json.</code></span>
+                        </p>
+                        <section id="webid-profile-data-model" inlist="" rel="schema:hasPart"
+                            resource="#webid-profile-data-model">
+                            <h3 property="schema:name" content="WebID Profile Data Model">Solid WebID Profile</h3>
+                            <div datatype="rdf:HTML" property="schema:description">
+                                <p>The <a href="#solid-profile">Solid WebID Profile</a> has the following properties:</p>
+                                <ul>
+                                    <li>One <code>rdf:type</code> property whose object is <code>foaf:Agent</code>.</li>
+                                    <li>One <code>space:preferencesFile</code>.</li>
```
Comment (elf-pavlik) [inline index.html:1188]: The namespaces section uses `pim` as a prefix, at the same time this line and the rest of the section use `space` as a prefix.

File: index.html [hunk 2]
```diff
@@ -1032,31 +1044,55 @@ <h3 property="schema:name">Namespaces</h3>
...
+                  </div>
+                </section>
+          
+                <section id="classes-of-products" inlist="" rel="schema:hasPart" resource="#classes-of-products" typeof="skos:ConceptScheme">
+                  <h4 property="schema:name skos:prefLabel">Classes of Products</h4>
+                  <div datatype="rdf:HTML" property="schema:description">
+                    <p property="skos:definition">The <span about="" rel="spec:classesOfProducts" resource="#classes-of-products">Solid WebID Profile</span> identifies the following <cite><a href="https://www.w3.org/TR/qaframe-spec/#cop-def" rel="dcterms:subject" resource="spec:ClassesOfProducts">Classes of Products</a></cite> for conforming implementations. These products are referenced throughout this specification.</p>
+          
+                    <dl rel="skos:hasTopConcept">
+                      <dt about="#ReaderApplication" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:Consumer" typeof="skos:Concept"><dfn id="ReaderApplication">ReaderApplication</dfn></dt>
```
Comment (elf-pavlik) [inline index.html:1074]:
```suggestion
                      <dt about="#Reader Application" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:Consumer" typeof="skos:Concept"><dfn id="ReaderApplication">ReaderApplication</dfn></dt>
```

File: index.html [hunk 3]
```diff
@@ -1032,31 +1044,55 @@ <h3 property="schema:name">Namespaces</h3>
...
+                      <dd about="#ReaderApplication" property="skos:definition">A <em>A <a href="https://solidproject.org/TR/protocol#solid-app">Solid app</a></em> that issues HTTP requests to consume content pertaining to WebID Profiles.</dd>
+                      <dt about="#WriterApplication" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:Consumer" typeof="skos:Concept"><dfn id="WriterApplication">ReaderApplication</dfn></dt>
```
Comment (elf-pavlik) [inline index.html:1076]:
```suggestion
                      <dt about="#WriterApplication" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:Consumer" typeof="skos:Concept"><dfn id="WriterApplication">Writer Application</dfn></dt>
```

File: index.html [hunk 4]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                            <span property="spec:statement"><a rel="spec:requirementSubject" href="#WriterApplication">Writer
+                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> update the <a
+                                        href="#solid-profile">Solid WebID Profile </a>by adding a triple of the form
+                                    <code>?webid rdfs:seeAlso ?extendedProfile.</code></span>
+                        </p>
+                        <div class="note" id="sameness-between-identities" inlist="" rel="schema:hasPart"
+                            resource="#sameness-between-identities">
+                            <h5 property="schema:name"><span>Note</span>:</h5>
+                            <div datatype="rdf:HTML" property="schema:description">
+                                <p>The use of <code>owl:sameAs</code> is encouraged when <a href="#use-cases">sameness between identities</a> is
```
Comment (elf-pavlik) [inline index.html:1302]: Given `?webid rdfs:seeAlso ?extendedProfile`, I see:

* `webid` denotes a non-informational resource
* `extendedProfile` denotes an informational resource

I don't see how `owl:sameAs` could be appropriate in this case.

**Comment by elf-pavlik** [comment]
I see a few issues related to #97 (and #63), which were discussed during last week's call. I think they can be resolved independently from this specific PR, I will just mention them here since they potentially would impact what comes in this PR.

* There seems to be an assumption that WebID Document (defined in WebID spec) is the same document as WebID Profile. I understand the requirement that WebID (Solid?) The profile would be hosted in Solid Storage, but I don't think it can be required that WebID (Solid?) The profile is the same as WebID Document, which doesn't have to be hosted in Solid Storage.
* As discussed in #97 other specifications should be able to extend WebID (Solid?) Profile. I think the Private Preferences should be a separate spec, similar to Type Indexes or possibly combined with Type Indexes. I see the Private Preferences section as very underspecified and appears to make some assumptions about authentication and authorization which may not always be correct. The following should be tracked as an open registry/directory rather than be included in the spec that will be snapshot. 
  * `solid:oidcIssuer`
  * `ldp:inbox`
  * `space:storage`
  * `space:preferencesFile`
* Mentioned registry/directory should also include if the property is protected, for example, `solid:oidcIssuer` or not.
* The concept of protected properties should be reevaluated altogether. Solid authorization seems to be on the resource, not the triple-based level. This seems especially challenging when there is no closed set of protected properties. If all properties by default are unprotected, a property that is intended to be protected ends up unprotected just because the server implementation didn't have it in the list of protected properties. The strategy of having separate resources for protected and unprotected statements seems more robust when it comes to extensibility. It also follows resource level access control.

File: index.html (hunk 2 above)
Comment (csarven) [reply index.html:1074]: ^ Not sure if the history shows but I've updated Pavlik's suggestion. I think what's intended is the text node `>Reader Application<` and not `about="#Reader Application"`.

File: index.html [hunk 5]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                        </div>
+                        <p about="" id="writer-application-webid-profile" rel="spec:requirement"
+                            resource="writer-application-webid-profile">
+                            <span property="spec:statement"><a rel="spec:requirementSubject" href="#WriterApplication">Writer
+                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use an
+                                an <code>HTTP PUT</code> or <code>PATCH</code> request targeting the Solid WebID Profile to be
+                                updated.</span>
+                        </p>
+                        <p about="" id="protected-properties" property="schema:description">
+                            Solid WebID Profile might include protected properties, such as <code>solid:oidcIssuer</code>.
```
Comment (elf-pavlik) [inline index.html:1252]: Since statements with the `solid:oidcIssuer` predicate is defined as *protected* in this spec, I assume it only applies to resources that are WebID Profiles. How the solid storage/resource server is going to determine that a given resource should be handled as a WebID Profile?

File: index.html (hunk 1 above)
Comment (csarven) [reply index.html:1188]: * N3 and Turtle serializations of http://www.w3.org/ns/pim/space use `ws` as the prefix for `http://www.w3.org/ns/pim/space#`
* http://prefix.cc/pim has an entry for the above, and http://prefix.cc/space has an entry for something entirely different.
* https://github.com/solid/solid-namespace/blob/7498b4d93b543c1c1508d2542013b97bd10db127/index.js#L51 uses `space`
* Not that this matters much but the source file is [space.n3](https://github.com/solid/vocab/blob/main/space.n3).
* Not that this matters much but I've been using `pim` although I do find `ws` or `space` relatively more clear because there are other vocabs under `pim/`.

Pick one at random :)

File: index.html (hunk 4 above)
Comment (csarven) [reply index.html:1302]: The note is not necessarily implying `webid sameAs extendedProfile` either, but I agree that it can can better indicate / emphasise on the URI that identifies the social agent and not the document regarding `owl:sameAs`.

Not sure if this is good enough but some paraphrasing can help (without going into depths about what sameAs is or how to use it properly - those that need something like that can make that call based on knowledge from elsewhere):

```suggestion
                                <p>The use of <code>owl:sameAs</code> is encouraged when <a href="#use-cases">sameness between identities</a> found in Profile documents is
```

File: index.html [hunk 6]
```diff
@@ -1032,31 +1044,55 @@ <h3 property="schema:name">Namespaces</h3>
...
+                    </dl>
+                  </div>
+                </section>
+          
+                <section id="interoperability" inlist="" rel="schema:hasPart" resource="#interoperability" typeof="skos:ConceptScheme">
+                  <h4 property="schema:name skos:prefLabel">Interoperability</h4>
+                  <div datatype="rdf:HTML" property="schema:description">
+                    <dl>
+          
+                      <dt about="#reader-application-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="reader-application-server-interoperability">Reader Application-Server interoperability</dfn></dt>
```
Comment (csarven) [inline index.html:1089]:
```suggestion
                      <dt about="#reader-application-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="reader-application-server-interoperability">Reader Application—Server interoperability</dfn></dt>
```

File: index.html [hunk 7]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                                                cases</a>.</p>
+                                    </div>
+                                </div>
+                            </div>
+                        </section>
+                    </div>
+        
+                    <h3 property="schema:name" content="Preferences Document">Preferences Document</h3>
+                    <div datatype="rdf:HTML" property="schema:description">
+                        <p>The <a href="#private-profile">Preferences Document</a> has the following properties:</p>
```
Comment (csarven) [inline index.html:1209]: Was `#preferences-document` or `#private-preferences` instead of `#private-profile` intended?

File: index.html [hunk 8]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                        <ul>
+                            <li>One <code>rdf:type</code> property whose object is <code>space:ConfigurationFile</code>.</li>
+                            <li>Zero, one, or more <code>rdfs:seeAlso</code>.</li>
+                        </ul>
+                    </div>
+                    <h3 property="schema:name" content="Extended Profile Documents">Extended Profile Documents</h3>
+                    <div datatype="rdf:HTML" property="schema:description">
+                        <p>The <a href="#extended-profile-documents">Extended Profile Documents</a> are also self-describing
```
Comment (csarven) [inline index.html:1217]:
```suggestion
                        <p>The <a href="#extended-profile-documents">Extended Profile Documents</a> are also <a href="https://www.w3.org/2001/tag/doc/selfDescribingDocuments">self-describing</a>
```

**Review by csarven — CHANGES_REQUESTED** [review]

**Comment by csarven** [comment]
>There seems to be an assumption that WebID Document (defined in WebID spec) is the same document as WebID Profile. I understand the requirement that WebID (Solid?) The profile would be hosted in Solid Storage, but I don't think it can be required that WebID (Solid?) The profile is the same as WebID Document, which doesn't have to be hosted in Solid Storage.

This is a bit hard to follow for me. Can you please paraphrase this?

As I see it from this specification and understanding in the community:

The Solid WebID Profile specification specifies the data model of the WebID Profile used in Solid, as well as any additional requirements to a Solid Protocol Server hosting the resource.

The ReaderApplication should be able to read any WebID Profile (conforming to WebID 1.0 ED) but the WriterApplication works against Solid Protocol's Server - to be further clear, WriteApplication is not intended to be able to update any WebID Profile out there, just what's on Solid storage.

>As discussed in
https://github.com/solid/webid-profile/issues/97 other specifications should be able to extend WebID (Solid?) Profile.

I don't have a particular objection to that proposal, after all, it is a legitimate design on its own. However, what I would emphasise on specifying (and requiring) is the minimal Solid WebID Profile Data Model - whatever that may be. If we split off too much with optional/modular specifications, there may not be much to say about the Solid WebID Profile at its core, and that frankly leaves this specification not a whole lot more than the WebID 1.0 ED. So, this is just trying to answer the age old question about what constitutes a Solid WebID Profile. Can a social agent use it to authenticate itself? store stuff at its storage? be contacted? Yes, I'm aware of various cases in which it may not be used for authentication, or to disclose a storage or use it to store anything, or be available for contact - and so the data shape is/should take that into account.

>Mentioned registry/directory should also include if the property is protected, for example, solid:oidcIssuer or not.
The concept of protected properties should be reevaluated altogether. Solid authorization seems to be on the resource, not the triple-based level. This seems especially challenging when there is no closed set of protected properties. If all properties by default are unprotected, a property that is intended to be protected ends up unprotected just because the server implementation didn't have it in the list of protected properties. The strategy of having separate resources for protected and unprotected statements seems more robust when it comes to extensibility. It also follows resource-level access control.

Again, no objection... but I think this can be leveraged by server using the shape specified by the specification to protect certain statements. The specifications needs to introduce how that shape association with the Solid WebID Profile document happens ( https://github.com/solid/webid-profile/pull/98#discussion_r1285636569 ). We have a data model / shape in the works, might as well apply that directly where we need it.

Aside: The above (explicit) approach can be used in the Solid Protocol for container / containment statements as well, but the interaction model or (cough) slash semantics already implicitly indicate that the resource is a container, so the server knows how to protect. We don't have anything equivalent to that for the Solid WebID Profile, so there needs to be a link relation referring to the shape.

File: index.html [hunk 9]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                            Solid WebID Profiles and share the same data model. <a href="reader-application">Reader
+                                Applications</a> retrieve representations of Extended Profile Documents the same way as they
+                            retrieve Solid WebID Profiles.</p>
+                    </div>
+                    <p about="" id="reader-application-extended-profile" rel="spec:requirement"
+                        resource="reader-application-extended-profile">
+                        <span property="spec:statement"><a rel="spec:requirementSubject" href="#ReaderApplication">Reader
+                                Application</a> <span rel="spec:requirementLevel" resource="spec:MAY">MAY</span> retrieve a
+                            representation of an <a href="#extended-profile-documents">Extended Profile Document</a>.</span>
+                    </p>
```
Comment (edwardsph) [inline index.html:1227]: As it stands, this is not a testable requirement as there is nothing to say what mechanism would be used.

File: index.html [hunk 10]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                            resource="#webid-profile-data-model">
+                            <h3 property="schema:name" content="WebID Profile Data Model">Solid WebID Profile</h3>
+                            <div datatype="rdf:HTML" property="schema:description">
+                                <p>The <a href="#solid-profile">Solid WebID Profile</a> has the following properties:</p>
+                                <ul>
+                                    <li>One <code>rdf:type</code> property whose object is <code>foaf:Agent</code>.</li>
+                                    <li>One <code>space:preferencesFile</code>.</li>
+                                    <li>Zero or more <code>ldp:inbox</code>.</li>
+                                    <li>Zero, one, or more <code>space:storage</code>.</li>
+                                    <li>Zero, one, or more <code>rdfs:seeAlso</code> to refer to related <a
```
Comment (edwardsph) [inline index.html:1191]: "one" is redundant - "Zero, or more" is the same thing

File: index.html [hunk 11]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                            </div>
+                        </section>
+                    </div>
+        
+                    <h3 property="schema:name" content="Preferences Document">Preferences Document</h3>
+                    <div datatype="rdf:HTML" property="schema:description">
+                        <p>The <a href="#private-profile">Preferences Document</a> has the following properties:</p>
+                        <ul>
+                            <li>One <code>rdf:type</code> property whose object is <code>space:ConfigurationFile</code>.</li>
+                            <li>Zero, one, or more <code>rdfs:seeAlso</code>.</li>
```
[the remark this answers is not in the document]
Comment (edwardsph) [reply index.html:1212]: As before, "one" is redundant - "Zero, or more" is the same thing

File: index.html [hunk 12]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                                        href="#solid-profile">Solid WebID Profile</a> before needing to use the <a
+                                        href="extended-profile-documents">Extended WebID Profile.</a></p>
+                            </div>
+                        </div>
+                        </div>
+                        <p about="" id="writer-application-webid-profile" rel="spec:requirement"
+                            resource="writer-application-webid-profile">
+                            <span property="spec:statement"><a rel="spec:requirementSubject" href="#WriterApplication">Writer
+                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use an
+                                an <code>HTTP PUT</code> or <code>PATCH</code> request targeting the Solid WebID Profile to be
```
Comment (edwardsph) [inline index.html:1248]:
```suggestion
                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use an
                                <code>HTTP PUT</code> or <code>PATCH</code> request targeting the Solid WebID Profile to be
```
"an" was repeated

File: index.html [hunk 13]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                                updated.</span>
```
Comment (edwardsph) [inline index.html:1249]: > Writer Application MUST use an HTTP PUT or PATCH request targeting the Solid WebID Profile to be updated

Ideally requirement statements should be standalone statements to make them easier to understand in other contexts such as test results. What I think is missing above is the clear purpose - the context is 'update' but perhaps it could be worded as "To update the Solid WebID Profile ...". Is there also something about the body and content type of such requests.

File: index.html [hunk 14]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                            resource="writer-application-create-extended-profile">
+                            <span property="spec:statement"><a rel="spec:requirementSubject" href="#WriterApplication">Writer
+                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use an
+                            <code>HTTP PUT</code> or <code>PATCH</code> request targeting the Extended Profile to be created.</span>
+                        </p>
+                        <p about="" id="writer-application-update-extended-profile" rel="spec:requirement"
+                            resource="writer-application-update-extended-profile">
+                            <span property="spec:statement"><a rel="spec:requirementSubject" href="#WriterApplication">Writer
+                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use an
+                             <code>HTTP PUT</code> or <code>PATCH</code> request targeting the Solid WebID Profile to be updated</span>
```
Comment (edwardsph) [inline index.html:1289]:
```suggestion
                             <code>HTTP PUT</code> or <code>PATCH</code> request targeting the Extended Profile to be updated</span>
```

**Review by edwardsph — COMMENTED** [review]
I mainly checked that the RDFa produced the correct requirements for the test harness and it is essentially correct. The line breaks in text do come through into the RDF and make it look odd e.g. 
```
  spec:statement """Reader
                                Application MAY retrieve a
                            representation of an Extended Profile Document."""@en .
```

Two other considerations are:
* Does the requirement work as a standalone statement? I think some are a bit too dependent on context for them to make sense but when they appear in test reports, they need to make sufficient sense to be understood without that context.
* Are requirements clearly testable -I have highlighted one that I think isn't as it stands

File: index.html (hunk 10 above)
Comment (TallTed) [reply index.html:1191]:
```suggestion
                                    <li>Zero or more <code>space:storage</code>.</li>
                                    <li>Zero or more <code>rdfs:seeAlso</code> to refer to related <a
```

File: index.html (hunk 6 above)
Comment (TallTed) [reply index.html:1089]:
```suggestion
                      <dt about="#reader-application-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="reader-application-server-interoperability">Reader Application&lt;–&gt;Server interoperability</dfn></dt>
```

File: index.html [hunk 15]
```diff
@@ -1032,31 +1044,55 @@ <h3 property="schema:name">Namespaces</h3>
...
+                  <h4 property="schema:name skos:prefLabel">Interoperability</h4>
+                  <div datatype="rdf:HTML" property="schema:description">
+                    <dl>
+          
+                      <dt about="#reader-application-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="reader-application-server-interoperability">Reader Application-Server interoperability</dfn></dt>
+                      <dd about="#reader-application-server-interoperability" property="skos:definition">Interoperability of implementations for <a href="#Server" rel="rdfs:seeAlso">Servers</a> and <a href="#ReaderApplication" rel="rdfs:seeAlso">Reader Application</a> is tested by evaluating an implementation’s ability to request and respond to HTTP messages that conform to this specification.</dd>
+                      <dt about="#writer-application-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="writer-application-server-interoperability">Writer Application-Server interoperability</dfn></dt>
```
[the remark this answers is not in the document]
Comment (TallTed) [reply index.html:1091]:
```suggestion
                      <dt about="#writer-application-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="writer-application-server-interoperability">Writer Application&lt;–&gt;Server interoperability</dfn></dt>
```

File: index.html [hunk 16]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                <section id="reading-profile" inlist="" rel="schema:hasPart" resource="#reading-profile">
+                    <h3 property="schema:name">Reading Profile</h3>
+                    <div datatype="rdf:HTML" property="schema:description">
+                        <p about="" id="reader-application-webid-profile" rel="spec:requirement"
+                            resource="reader-application-webid-profile">
+                            <span property="spec:statement">To retrieve a representation of a Solid WebID Profile, <a
+                                    rel="spec:requirementSubject" href="#ReaderApplication">Reader Application</a> <span
+                                    rel="spec:requirementLevel" resource="spec:MUST">MUST</span> MUST make
+                                an <code>HTTP GET</code> request targeting a Solid WebID Profile resource including a
+                                <code>Content-Type header</code> requesting a representation in <code>text/turtle</code> or
```
Comment (TallTed) [inline index.html:1178]:
```suggestion
                                <code>Content-Type</code> header requesting a representation in <code>text/turtle</code> or
```

File: index.html (hunk 11 above)
[the remark this answers is not in the document]
Comment (TallTed) [reply index.html:1212]:
```suggestion
                            <li>Zero or more <code>rdfs:seeAlso</code> properties with tangential relation, lacking any more specific relation, to the entity denoted by the Solid WebID.</li>
```

File: index.html (hunk 9 above)
Comment (TallTed) [reply index.html:1227]: As a MAY, it would typically not be tested, so it doesn't matter if it's testable. That said, the mechanism would typically be dictated by the URL of the Extended Profile Document, and any representation returned by the server would be acceptable.

File: index.html (hunk 12 above)
Comment (TallTed) [reply index.html:1248]:
```suggestion
                                    Application</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use an
                                HTTP <code>PUT</code> or <code>PATCH</code> request targeting the Solid WebID Profile to be
```

File: index.html [hunk 17]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+            <div datatype="rdf:HTML" property="schema:description">
+                <section id="reading-profile" inlist="" rel="schema:hasPart" resource="#reading-profile">
+                    <h3 property="schema:name">Reading Profile</h3>
+                    <div datatype="rdf:HTML" property="schema:description">
+                        <p about="" id="reader-application-webid-profile" rel="spec:requirement"
+                            resource="reader-application-webid-profile">
+                            <span property="spec:statement">To retrieve a representation of a Solid WebID Profile, <a
+                                    rel="spec:requirementSubject" href="#ReaderApplication">Reader Application</a> <span
+                                    rel="spec:requirementLevel" resource="spec:MUST">MUST</span> MUST make
+                                an <code>HTTP GET</code> request targeting a Solid WebID Profile resource including a
```
Comment (TallTed) [inline index.html:1177]:
```suggestion
                                an HTTP <code>GET</code> request targeting a Solid WebID Profile resource including a
```

File: index.html [hunk 18]
```diff
@@ -1127,6 +1163,150 @@ <h2 property="schema:name">Discovering a complete Solid profile</h2>
...
+                                updated.</span>
+                        </p>
+                        <p about="" id="protected-properties" property="schema:description">
+                            Solid WebID Profile might include protected properties, such as <code>solid:oidcIssuer</code>.
+                        </p>
+                        <p about="" id="server-update-protected-triples" rel="spec:requirement"
+                            resource="server-update-protected-triples">
+                            <span property="spec:statement"><a rel="spec:requirementSubject" href="#Server">Servers</a> <span
+                                    rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> allow
+                                <code>HTTP PUT</code> or <code>PATCH</code> on a Solid WebID Profile to update its protected
```
Comment (TallTed) [inline index.html:1258]:
```suggestion
                                HTTP <code>PUT</code> or <code>PATCH</code> on a Solid WebID Profile to update its protected
```

File: index.html (hunk 14 above)
Comment (TallTed) [reply index.html:1289]:
```suggestion
                             HTTP <code>PUT</code> or <code>PATCH</code> request targeting the Extended Profile to be updated</span>
```

File: index.html (hunk 4 above)
Comment (VirginiaBalseiro) [reply index.html:1302]: Would something like this work:

> While the rdfs:seeAlso property is recommended to establish links to extended profiles or additional information, the
use of the owl:sameAs property is encouraged when connecting two WebIDs that denote the same identity, as shown in the example: `?webid owl:sameAs ?webid.`

File: index.html (hunk 11 above)
[the remark this answers is not in the document]
Comment (VirginiaBalseiro) [reply index.html:1212]: How about this? 

```suggestion
                            <li>Zero or more <code>rdfs:seeAlso</code> to refer to extended profiles or include tangentially related documents without specific relation to the entity denoted by the Solid WebID.</li>
```

File: index.html (hunk 9 above)
Comment (VirginiaBalseiro) [reply index.html:1227]: > here is nothing to say what mechanism would be used.

Just before that requirement, the text says:

> Reader Applications retrieve representations of Extended Profile Documents the same way as they retrieve Solid WebID Profiles.

Is this enough to clarify the mechanism?

File: index.html (hunk 4 above)
Comment (TallTed) [reply index.html:1302]: (Everything is more complicated than it seems at first glance...)

File: index.html (hunk 6 above)
Comment (csarven) [reply index.html:1089]: I agree with the use of `–` (en dash) instead of `—` (em dash). I updated my suggestion. I don't find `&lt;` and `&gt;` necessary.

File: index.html (hunk 4 above)
Comment (csarven) [reply index.html:1302]: I would even suggest adding what an understanding may be between: one entity referring to another vs. both entities referring to each other. This generally ties to the other use cases, e.g., verification.

File: index.html (hunk 7 above)
Comment (TallTed) [reply index.html:1209]: Good question. I don't think that originated with me, so this should probably be asked outside of my review.

File: index.html (hunk 6 above)
Comment (TallTed) [reply index.html:1089]: OK

File: index.html (hunk 15 above)
[the remark this answers is not in the document]
Comment (TallTed) [reply index.html:1091]: OK

**Review by TallTed — CHANGES_REQUESTED** [review]
Minor

File: index.html (hunk 1 above)
Comment (csarven) [reply index.html]: Let's stick to `pim` for now. We're still in drafts land... can come back to this and other prefixes across all specs if we need to.

File: index.html (hunk 13 above)
Comment (csarven) [reply index.html]: Writer Application is a Solid app (Solid Protocol). But we need to revisit this with respect to the Solid Protocol because it is not currently defined as an independent product class. Writer Application could be a specialisation of Solid Protocol Client but then that would still require us to revisit Solid app in Solid Protocol. The general understanding is that the Solid app issues HTTP requests through the Solid Protocol Client against a Solid Protocol Server. So, Writer Application writes to a Solid WebID Profile on a Solid storage.

Solid WebID Profile is an RDF document. Solid Servers can potentially accept any concrete RDF syntax, but only required to respond to Turtle and JSON-LD. The body of the request includes an RDF representation based on the Solid WebID Profile data model.

---

I agree with the suggestion to have multiple (identifiable) requirements. And doing that for all similar requirements. (It is also okay to leave it as is for the time being - not super urgent per se.)

---

Purpose at the front instead of at the end seems okay too. I would only suggest that the specification should generally be consistent (and boring), so if "purpose" comes first, do it for all similar requirements, e.g., "[purpose] [requirementSubject] [requirementLevel] [behaviour]".

We can look at this from the Solid QA end and make a proposal that could be used across all specs.

File: index.html (hunk 9 above)
Comment (csarven) [reply index.html]: The text before the requirement seems to serve this purpose.

File: index.html (hunk 5 above)
Comment (VirginiaBalseiro) [reply index.html]: Created https://github.com/solid/webid-profile/issues/100 to follow up on this.

File: index.html (hunk 13 above)
Comment (VirginiaBalseiro) [reply index.html]: I created https://github.com/solid/webid-profile/issues/101 to follow up on this.
